src/main/scala/cromwell/engine/workflow/ValidateActor.scala

```diff
@@ -0,0 +1,55 @@
+package cromwell.engine.workflow
+
+import akka.actor.{Actor, Props}
+import com.typesafe.scalalogging.LazyLogging
+import cromwell.binding._
+import cromwell.webservice.PerRequest.RequestComplete
+import cromwell.webservice.{WorkflowJsonSupport, WorkflowValidateResponse}
+import spray.http.StatusCodes
+import spray.httpx.SprayJsonSupport._
+import spray.json._
+
+import scala.concurrent.Future
+import scala.language.postfixOps
+import scala.util.{Failure, Success}
+
+object ValidateActor {
+  sealed trait ValidateActorMessage
+  case object ValidateWorkflow extends ValidateActorMessage
+
+  def props(wdlSource: WdlSource, wdlJson: WdlJson): Props = {
+    Props(new ValidateActor(wdlSource, wdlJson))
+  }
+}
+
+class ValidateActor(wdlSource: WdlSource, wdlJson: WdlJson)
+  extends Actor with LazyLogging {
+
+  import WorkflowJsonSupport._
+  import context.dispatcher
+  import ValidateActor.ValidateWorkflow
+  private val tag = "ValidateActor"
+
+  override def receive = {
+    case ValidateWorkflow =>
+      val sentBy = sender()
+      logger.info(s"$tag for $sentBy")
+      val futureInputs = for {
+        eventualNamespace <- Future(NamespaceWithWorkflow.load(wdlSource, WorkflowManagerActor.BackendType))
+        inputs <- Future(wdlJson.parseJson) map {_.asJsObject.fields}
```

Comment:
Reverted to `.map()` as intellij wanted to line wrap the curly braces with spaces.